node/src/protocol/triple.rs

```diff
@@ -166,7 +175,7 @@ impl TripleManager {
             max_triples,
         } = self.triple_cfg;
         let not_enough_triples =
-            || self.my_len() < min_triples && self.potential_len() < max_triples;
+            || self.my_len() < min_triples && self.potential_len() - self.len() < max_triples;
```

Comment:
so doing `- self.len()` will get us `self.generators.len()`, which would mean that we would always be generating and have a max_triples amount of generators alive. Don't think we should do this `max_triples` should determine the total amount of triples in the whole network

Reply:
no this only generates when both mine < min_triples and total generators < max triples. This is necessary because there's an edge case where mine = 1 and total generated >= 10, this condition will successfully break that edge case.

Reply:
So there's an edge case with mine = 1 and total = 10, and we will never generate any more triples for this case. And since presignatures need 2 mine, this 1 triple will never be used. I was doing above change to break this. But I guess a more reasonable way would actually be something like my_generator_len()

Reply:
I think we should have a different config then for generators, something like `max_concurrent_triples`, where
`min_triple < max_concurrent_triples < max_triples`. So `max_triples` still determines the total supply of triples in the network

Reply:
I changed condition to `(self.my_len() < min_triples && self.potential_len() == self.len())
                || self.potential_len() < max_triples`.
Basically generate triple when mine < min_triples and no triple protocol ongoing, or when existing + ongoing triples < max_triples.